store/sqlstore/oauth_store.go

```diff
@@ -161,139 +163,139 @@ func (as SqlOAuthStore) DeleteApp(id string) *model.AppError {
+func (as SqlOAuthStore) GetAccessData(token string) (*model.AccessData, error) {
 	accessData := model.AccessData{}
 
 	if err := as.GetReplica().SelectOne(&accessData, "SELECT * FROM OAuthAccessData WHERE Token = :Token", map[string]interface{}{"Token": token}); err != nil {
-		return nil, model.NewAppError("SqlOAuthStore.GetAccessData", "store.sql_oauth.get_access_data.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrapf(err, "failed to get OAuthAccessData with token=%s", token)
 	}
 	return &accessData, nil
 }
 
-func (as SqlOAuthStore) GetAccessDataByUserForApp(userId, clientId string) ([]*model.AccessData, *model.AppError) {
+func (as SqlOAuthStore) GetAccessDataByUserForApp(userId, clientId string) ([]*model.AccessData, error) {
 	var accessData []*model.AccessData
 
 	if _, err := as.GetReplica().Select(&accessData,
 		"SELECT * FROM OAuthAccessData WHERE UserId = :UserId AND ClientId = :ClientId",
 		map[string]interface{}{"UserId": userId, "ClientId": clientId}); err != nil {
-		return nil, model.NewAppError("SqlOAuthStore.GetAccessDataByUserForApp", "store.sql_oauth.get_access_data_by_user_for_app.app_error", nil, "user_id="+userId+" client_id="+clientId, http.StatusInternalServerError)
+		return nil, errors.Wrapf(err, "failed to delete OAuthAccessData with userId=%s and clientId=%s", userId, clientId)
 	}
 	return accessData, nil
 }
 
-func (as SqlOAuthStore) GetAccessDataByRefreshToken(token string) (*model.AccessData, *model.AppError) {
+func (as SqlOAuthStore) GetAccessDataByRefreshToken(token string) (*model.AccessData, error) {
 	accessData := model.AccessData{}
 
 	if err := as.GetReplica().SelectOne(&accessData, "SELECT * FROM OAuthAccessData WHERE RefreshToken = :Token", map[string]interface{}{"Token": token}); err != nil {
-		return nil, model.NewAppError("SqlOAuthStore.GetAccessData", "store.sql_oauth.get_access_data.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrapf(err, "failed to find OAuthAccessData with refreshToken=%s", token)
 	}
 	return &accessData, nil
 }
 
-func (as SqlOAuthStore) GetPreviousAccessData(userId, clientId string) (*model.AccessData, *model.AppError) {
+func (as SqlOAuthStore) GetPreviousAccessData(userId, clientId string) (*model.AccessData, error) {
 	accessData := model.AccessData{}
 
 	if err := as.GetReplica().SelectOne(&accessData, "SELECT * FROM OAuthAccessData WHERE ClientId = :ClientId AND UserId = :UserId",
 		map[string]interface{}{"ClientId": clientId, "UserId": userId}); err != nil {
-		if strings.Contains(err.Error(), "no rows") {
+		if err == sql.ErrNoRows {
 			return nil, nil
 		}
-		return nil, model.NewAppError("SqlOAuthStore.GetPreviousAccessData", "store.sql_oauth.get_previous_access_data.app_error", nil, err.Error(), http.StatusNotFound)
+		// review: I will keep the notFound status but I think we should return 500 here
```

Comment:
Yes, totally agree, the should be a 500 error.

Reply:
I would say we can do the change here itself, or it's also fine if you want to send a PR later. Up to you.